components/Feed/Post/PostHeader.tsx

```diff
@@ -20,6 +21,28 @@ type PostHeaderProps = {
 
 const PostHeader = ({id, userImage, name, authorUid, email, timestamp}: PostHeaderProps) => {
     const [user] = useAuthState(auth);
+    const [numLikes, setNumLikes] = useState(0);
+    
+    // Use useEffect to bind on document loading the
+    // function that will set the number of likes on
+    // each change of the DB (triggered by onSnapshot)
+    useEffect(() => {
+        db.collection("posts")
+        .doc(id)
+        .onSnapshot((snapshot) => {
+            // Get the likes map
+            const likesMap = snapshot.data().likes;
+    
+            // Count the entries that are True
+            let ctr = 0;
+            for (const [key, value] of Object.entries(likesMap)) {
```

Comment:
This throws on posts without the likesMap object.
We need to modify NewPostForm (line 211 - postData) to add an empty object "likes".

Reply:
Added.